Testing on x86_64-darwin, fails to compile fuzz tests.

```
[ 86%] Building CXX object test/fuzz/CMakeFiles/fuzz_bounds.dir/bounds.cpp.o
[ 86%] Linking CXX executable runtime_internal_memory_arena
error: current translation unit is compiled with the target feature '-fsanitize=fuzzer-no-link' but the AST file was not
1 error generated.
make[2]: *** [test/fuzz/CMakeFiles/fuzz_bounds.dir/build.make:78: test/fuzz/CMakeFiles/fuzz_bounds.dir/bounds.cpp.o] Error 1
```

Post-release, upstream committed https://github.com/halide/Halide/commit/09c5d1d19ec8e6280ccbc01a8a12decfb27226ba to disable these tests unless specifically requested. Prolly either `fetchpatch` to change the default ahead of the next release or set `WITH_TEST_FUZZ` to off.

--- a/pkgs/development/compilers/halide/default.nix
+++ b/pkgs/development/compilers/halide/default.nix
@@ -67,11 +67,8 @@ stdenv.mkDerivation rec {
 
   # Note: disable mullapudi2016_fibonacci because it requires too much
   # parallelism for remote builders
-  ctestArgs = "--output-on-failure -E 'mullapudi2016_fibonacci'";
-  checkPhase = ''
-    runHook preCheck
-    ctest ${ctestArgs}
-    runHook postCheck
+  preCheck = ''
+    checkFlagsArray+=("ARGS=-E 'mullapudi2016_fibonacci'")
   '';
 
   # Note: only openblas and not atlas part of this Nix expression

--- a/pkgs/development/compilers/halide/default.nix
+++ b/pkgs/development/compilers/halide/default.nix
@@ -57,6 +57,10 @@ stdenv.mkDerivation rec {
     "-DTARGET_WEBASSEMBLY=OFF"
     # Disable performance tests since they may fail on busy machines
     "-DWITH_TEST_PERFORMANCE=OFF"
+    # Disable fuzzing tests -- this has become the default upstream after the
+    # v16 release (See https://github.com/halide/Halide/commit/09c5d1d19ec8e6280ccbc01a8a12decfb27226ba)
+    # These tests also fail to compile on Darwin because of some missing command line options...
+    "-DWITH_TEST_FUZZ=OFF"
   ];
 
   doCheck = true;

